bonus.md

```diff
@@ -3,11 +3,17 @@
 Edit this file to answer the following questions. Give examples. You may need to do a bit of research. :)
 
 1. Why is Hash.fetch useful?
+Its useful because it can raise an error if a matching Key isn't found. This would make it easier to troubleshoot an incorrect argument passing.
 
 2. What is the disadvantage of single inheritance, and what's a possible solution to that problem?
+Some methods may be very similar for children of a parent class, yet still enough differentiation exists. Another potential problem is that inheritance in ruby is linear (Though this can also be a positive). It will only go up the chain to the next ancestor to locate a solution.
+I believe you could potentially use either constants to get around this.
 
 3. Why are these classes initialized with an options hash?
+This allows a default value to be passed in if no arguments are given. You can still instantiate a class and pass it no arguments, and it will default to an empty hash, or nil in this case.
 
 4. What is the purpose of the private keyword in a class? What does it protect you from and why is that valuable?
+This allows you to keep certain class data private from public facing audiences. The class is the only part of the program that has access to the private methods and data.
 
```

Comment:
Not really. It's very easy to access a private method from outside a class, with obj.send(:private_method_name). It's really more to signal to clients of your class that these are methods that are internal-only and they shouldn't need to call them. Everything about the private section is part of the public interface, and users are expected to call those methods.